src/nsrt_learning/side_predicate_learning.py

```diff
@@ -456,18 +479,67 @@ def _sideline(self) -> List[PartialNSRTAndDatastore]:
                     for a in pnad.op.preconditions
                 }
 
+        # Now that the add effects and preconditions are correct,
+        # make a list of all final PNADs. Note
+        # that these final PNADs only come from the
+        # param_opt_to_nec_pnads dict, since we can be assured
+        # that our backchaining process ensured that the
+        # PNADs in this dict cover all of the data!
+        all_pnads = []
+        for pnad_list in param_opt_to_nec_pnads.values():
+            for pnad in pnad_list:
+                all_pnads.append(pnad)
+
         # At this point, all PNADs have correct parameters, preconditions,
         # and add effects. We now finalize the delete effects and side
         # predicates. Note that we have to do delete effects first, and
         # then side predicates, because the latter rely on the former.
-        for pnad in param_opt_to_pnad.values():
+        for pnad in all_pnads:
             self._finalize_pnad_delete_effects(pnad)
             self._finalize_pnad_side_predicates(pnad)
 
         # Before returning, sanity check that harmlessness holds.
-        final_pnads = list(param_opt_to_pnad.values())
-        assert self._check_harmlessness(final_pnads)
-        return final_pnads
+        assert self._check_harmlessness(all_pnads)
+        return all_pnads
+
+    @staticmethod
+    def _find_unification(
+        necessary_add_effects: Set[GroundAtom],
+        pnad: PartialNSRTAndDatastore,
+        segment: Segment,
+        find_partial_grounding: bool = False
+    ) -> Optional[_GroundSTRIPSOperator]:
+        """Find a mapping from the variables in the PNAD add effects and option
+        to the objects in necessary_add_effects and the segment's option.
+
+        If one exists, we don't need to modify this PNAD. Otherwise, we
+        must make its add effects more specific. Note that we are
+        assuming all variables in the parameters of the PNAD will appear
+        in either the option arguments or the add effects. This is in
+        contrast to strips_learning.py, where delete effect variables
+        also contribute to parameters. If find_partial_grounding is
```

Comment:
can we rename `find_partial_grounding` to something that makes clear this superset/subset behavior?